### PR TITLE
Remove a ton of // back-compat comments (and back-compat code)

### PR DESCRIPTION
**Pull request by vladsud**
Did not remove all of it - there were some places that
- were added recent and did not bake enough
- I was not sure if it's actually save to do changes comments suggested to do
- on one or two places comments were not valid / hard to follow instructions.

### CODE REVIEW

File: packages/loader/container-loader/src/container.ts [hunk 1]
```diff
@@ -602,10 +597,8 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
                 // This should be fired after taking the summary because it is the place where we are
                 // starting to attach the container to storage.
                 // Also, this should only be fired in detached container.
-                if (this.context.runtimeVersion !== undefined && this.context.runtimeVersion >= "0.25") {
```
Comment (curtisman) [inline packages/loader/container-loader/src/container.ts:605]: Good that we remove it. Noting that string based version comparison doesn't work,  and we should think about how to make sure this kind of bugs doesn't get into the code base

**Review by curtisman — APPROVED** [review]

File: packages/loader/container-loader/src/deltaManager.ts [hunk 2]
```diff
@@ -1375,8 +1375,6 @@ export class DeltaManager
         }
         this.baseTerm = message.term;
 
-        this.emit("beforeOpProcessing", message);
```
Comment (tanviraumi) [inline packages/loader/container-loader/src/deltaManager.ts:1378]: Whats a good alternative for "beforeOpProcessing"? I have been using it for measuring E2E op latency from a driver perspective (before the op actually processed by runtime)?

**Review by tanviraumi — APPROVED** [review]
Left a comment on removal of "beforeOpProcessing"

File: packages/loader/container-loader/src/deltaManager.ts (hunk 2 above)
Comment (vladsud) [reply packages/loader/container-loader/src/deltaManager.ts:1378]: the "op" event's 2nd argument is time it took to process op by runtime (+ deltamanager overhead). 
```
        const endTime = Date.now();
        this.emit("op", message, endTime - startTime);
```
Is this sufficient?

**Review by agarwal-navin — APPROVED** [review]

File: packages/loader/container-loader/src/deltaManager.ts (hunk 2 above)
Comment (tanviraumi) [reply packages/loader/container-loader/src/deltaManager.ts:1378]: Yes that should be sufficient for our use case.
